EpsilonAuxBMS/Inc/SetContactorsTask.h

```diff
@@ -5,6 +5,38 @@
 
 extern ADC_HandleTypeDef hadc1; // main.c
 
+//structs
+typedef enum ContactorsSettingState ContactorsSettingState;
+typedef struct SetContactorStateInfo SetContactorStateInfo ;
+
 // Task for turning on contactors
 // arg: osMutexId* auxStatusMutex
-void setContactorsTask(void const* arg);
\ No newline at end of file
+void setContactorsTask(void const* arg);
+// Functions for state's logic.
+void firstCheck(SetContactorStateInfo* stateInfo);
+void commonEnableCheck(SetContactorStateInfo* stateInfo);
+void chargeEnableCheck(SetContactorStateInfo* stateInfo);
+void dischargeEnableCheck(SetContactorStateInfo* stateInfo);
+void idle(SetContactorStateInfo* stateInfo);
+void blocked(SetContactorStateInfo* stateInfo);
+
+// Checks if stateInfo->hasDischargeBeenSet is set but discharge is disconnected
+const uint8_t dischargeDisconnectedDuringContactorSetting(const uint8_t dischargeSense, uint8_t hasDischargeBeenSet);
+
+// Checks if stateInfo->hasChargeBeenSet is set but charge disconnected
+const uint8_t chargeDisconnectedDuringContactorSetting(const uint8_t chargeSense, uint8_t hasChargeBeenSet);
+
+// Checks if discharge contactor can be turn on
+const uint8_t canTurnOnDishargeContactor(const uint8_t dischargeSense);
+
+// Checks if charge contactor can be turn on
+const uint8_t canTurnOnChargeContactor(const uint8_t chargeSense);
+
+// Checks if the discharge contactor should be on, but isn't
+const uint8_t dischargeDisconnectedUnexpectedly(const uint8_t dischargeSense, const uint8_t isdischargeTurningOn);
+
+// Checks if the charge contactor should be on, but isn't
+const uint8_t chargeDisconnectedUnexpectedly(const uint8_t chargeSense, const uint8_t isChargeTurningOn);
+
+//First check before setting states in case battery voltages are not in range, start up sequence is not done and makes sure the state is not disconnected
+const uint8_t safeToSetContactor(ContactorsSettingState state);
```

Comment:
@bill-luu Just to confirm. Did you want the prototypes and implementation in completely new source and header files?

Reply:
States should be in different files, and the other functions should be moved accordingly to who uses it.

Reply:
@bill-luu By separate files, do you mean that there should be a file for each state? Or just one file that has all the states and then the helper functions in a separate file?

Reply:
I guess I forgot to answer this question